perun-core/src/main/java/cz/metacentrum/perun/core/api/MembersManager.java

```diff
@@ -1506,7 +1507,23 @@ List<Map<String, String>> createSponsoredMembersFromCSV(PerunSession sess, Vo vo
 	 * @throws GroupNotExistsException if there is no such query group
 	 * @throws PrivilegeException insufficient permission
 	 */
-	Paginated<RichMember> getMembersPage(PerunSession sess, Vo vo, MembersPageQuery query, List<String> attrNames) throws VoNotExistsException, PrivilegeException, GroupNotExistsException;
+	Paginated<RichMember> getMembersPage(PerunSession sess, Vo vo, MembersPageQuery query, List<String> attrNames) throws VoNotExistsException, PrivilegeException, GroupNotExistsException, PolicyNotExistsException;
+
+	/**
+	 * Get page of members from the given vo, with the given attributes, based on policy.
+	 *
+	 * @param sess session
+	 * @param vo vo
+	 * @param query query with page information
+	 * @param attrNames attribute names
+	 * @param policy policy to use
+	 * @return page of requested rich members
+	 * @throws VoNotExistsException if there is no such vo
+	 * @throws GroupNotExistsException if there is no such query group
+	 * @throws PrivilegeException insufficient permission
+	 */
+	Paginated<RichMember> getMembersPage(PerunSession sess, Vo vo, MembersPageQuery query, List<String> attrNames, String policy) throws VoNotExistsException, PrivilegeException, GroupNotExistsException, PolicyNotExistsException;
```

Comment:
Just a question, as I am not too aware of the task's context. Do we need the policy argument? As I understand, the default policy (`filter-getMembersPage_policy`) should be the thing that determines how members are filtered, and admins of each instance can configure this as they please.

Reply:
I used the policy argument for easier handling of the tests. It is not strictly needed for the task, but it makes the tests more accessible and easier to use.

Reply:
I see, it's fine then